integration-tests/actions.go

```diff
@@ -616,3 +616,83 @@ func (tr TestRun) unbondTokens(
 		log.Fatal(err, "\n", string(bz))
 	}
 }
+
+type SlashAction struct {
+	downOn chainID
+	toDown validatorID
+}
+
+func (tr TestRun) InvokeSlash(action SlashAction, verbose bool) {
+	tr.ToggleValidatorDowntime(action.downOn, action.toDown, true, verbose)
+	// Wait appropriate amount of blocks for validator to be slashed
+	tr.waitBlocks(action.downOn, 10, time.Minute)
+}
+
+type RestoreVotingPowerAction struct {
+	bringUpOn chainID
+	provider  chainID
+	toRestore validatorID
+}
+
+func (tr TestRun) RestoreVotingPower(action RestoreVotingPowerAction, verbose bool) {
+	tr.ToggleValidatorDowntime(action.bringUpOn, action.toRestore, false, verbose)
+	// Wait the downtime_jail_duration set in config.go
+	time.Sleep(61 * time.Second)
```

Comment:
This too

Reply:
I had to move this line back to the other action, ```unjailValidator```, because the wait time isn't effective until the validator jailing is actually processed by the provider (after a slash packet is relayed from consumer -> provider).

Alternatively I can relay a packet within ```downtimeSlashAction```, but that's not consistent with other actions